lua_examples/luaOTA/check.lua

```diff
@@ -0,0 +1,66 @@
+--SAFETRIM
+--------------------------------------------------------------------------------
+-- LuaOTA provisioning system for ESPs using NodeMCU Lua
+-- LICENCE: http://opensource.org/licenses/MIT
+-- TerryE  15 Jul 2017
+--
+-- See luaOTA.md for description and implementation notes
+
+--------------------------------------------------------------------------------
+
+-- upvals
+local crypto, file, json,  net, node, table, tmr, wifi =
+      crypto, file, sjson, net, node, table, tmr, wifi
+local error, pcall   = error, pcall 
+local loadfile, gc   = loadfile, collectgarbage
+local concat, unpack = table.concat, unpack or table.unpack 
+
+local self = {post = node.task.post, prefix = "luaOTA/", conf = {}}
+
+self.log     = (DEBUG == true) and print or function() end 
```

Comment:
Nick, not sure what you mean.  I use an explicit `print` when I want to log the message with `DEBUG` false or nil.  This is just a lot simpler (and less code size) than having lots of `if DEBUG then print ( . . . ) end` statements in the code.  Its actually even faster to have 
```Lua
self.log     = (DEBUG == true) and print or type
```
as `type()` is a base C library function which accepts any argument list and returns a string (which is then ignored), but this would _really_ confuse most Lua beginners.